internal/gcsx/bucket_manager.go

```diff
@@ -212,6 +212,9 @@ func (bm *bucketManager) SetUpBucket(
 	// Enable content type awareness
 	b = NewContentTypeBucket(b)
 
+	// Enable monitoring
+	b = NewMonitoringBucket(b)
```

Comment:
Only if monitoring is enabled?

Reply:
The current switch only controls whether the metrics are exported. The metrics will always be counted internally. The cost of keeping them is negligible. 

If we decide to use one switch to turn off all the metrics, although this might be easy to implement now since all the metrics exist in one place (monitoring_bucket.go), it can be quite complicated in the future when we have metrics scattered around different files. That means the value of the switch has to be passed around, and "if-else" clauses executed everywhere to decide whether the metrics should be counted, which can be quite ugly.

What do you think? Is it necessary to have all the internal metrics as optional?

Reply:
Good points. Let me think about it.